extensions/kubernetes-client/deployment/src/main/java/io/quarkus/kubernetes/client/deployment/KubernetesClientProcessor.java

```diff
@@ -87,8 +87,14 @@ public void process(ApplicationIndexBuildItem applicationIndex, CombinedIndexBui
                 .toArray(String[]::new);
         reflectiveClasses.produce(new ReflectiveClassBuildItem(true, false, deserializerClasses));
 
-        reflectiveClasses
-                .produce(new ReflectiveClassBuildItem(true, false, "io.fabric8.kubernetes.api.model.IntOrString"));
```

Comment:
Is it normal that you remove this one? It doesn't implement `JsonSerializer` AFAICS.

Reply:
OK, I wanted this patch and wasn't sure removing this one was a good idea. So I decided to be on the safe side and amend your commit to reinject it.

We can discuss that later.

Reply:
Sorry for the late reply, yes, it's normal, this was added in PR https://github.com/quarkusio/quarkus/pull/3349 and released in _0.20.0_, but it don't actually solves the issue since it register the main class, not the inner **Serializer** class (see the error: `Caused by: java.lang.IllegalArgumentException: Class io.fabric8.kubernetes.api.model.IntOrString$Serializer has no default (no arg) constructor`)

The main classes of `IntOrString` and `Quantity` do not implement directly the JsonSerializer/JsonDeserializer, they use an inner static class (https://github.com/fabric8io/kubernetes-client/search?q=extends+JsonSerializer&unscoped_q=extends+JsonSerializer)

But sure there is no problem with leaving it there...

Reply:
I'll ask @geoand when he's back if we can safely remove it. Let's be safe for now as I will release in a few hours :).

Reply:
It's hard to tell without me looking at the actual Kubernetes Client code.

+1 for your proposed approach. I'll take a deeper look when I get back next week

Reply:
@geoand go away from that keyboard :)